client/api.go

```diff
@@ -27,14 +28,21 @@ func HostURL() *url.URL {
 }
 
 func hostURL(urlStr string) *url.URL {
+	apiUrl := os.Getenv("FN_API_URL")
+	currentContext := viper.Get(config.CurrentContext)
+	if apiUrl != "" && currentContext != "" {
+		fmt.Println("Warning: environment variable FN_API_URL is overriding configured context property api_url from current-context")
```

Comment:
Here and elsewherre I think errors/warnings should go to stderr: 

```
fmt.Fprintf(os.Stderr,....
```

Reply:
+1 on routing warnings to stderr -  @shaunsmith would like the ability to output results of 'list' operations as JSON and parse the output, so a warning in there would pollute the output.

More broadly though, as commented elsewhere in #249, I wonder whether we should stop allowing the mixing of contexts and env vars - use one or the other. Leave context unset to use env vars, or use a context in which case env vars are ignored.

Reply:
Would that require an `fn context unset` command in the case they have a context set but want to revert back using envvars? Or do we just expect them to edit the config.yaml to remove current-context?